Examples/Swift/AppDelegateCarPlay.swift

```diff
@@ -0,0 +1,99 @@
+import UIKit
+import MapboxNavigation
+#if canImport(CarPlay)
+import CarPlay
+import MapboxCoreNavigation
+import MapboxDirections
+
+
+@available(iOS 12.0, *)
+extension AppDelegate: CPApplicationDelegate {
+    
+    // MARK: CPApplicationDelegate
+    
+    func application(_ application: UIApplication, didConnectCarInterfaceController interfaceController: CPInterfaceController, to window: CPWindow) {
+        CarPlayManager.shared.delegate = self
+        CarPlayManager.shared.application(application, didConnectCarInterfaceController: interfaceController, to: window)
+    }
+    
+    func application(_ application: UIApplication, didDisconnectCarInterfaceController interfaceController: CPInterfaceController, from window: CPWindow) {
+        CarPlayManager.shared.delegate = nil
+        CarPlayManager.shared.application(application, didDisconnectCarInterfaceController: interfaceController, from: window)
+    }
+}
+
+@available(iOS 12.0, *)
+extension AppDelegate: CarPlayManagerDelegate {
+    
+    // MARK: CarPlayManagerDelegate
+    func carPlayManager(_ carPlayManager: CarPlayManager, didBeginNavigationWith routeController: RouteController) {
```

Comment:
This code (and didEnd...) is specific to the drop-in UI -- can we move it somewhere else, perhaps into the `NavigationViewController` itself?

Reply:
Tried to move this over to NVC but because this happens before the NVC is in the view hierarchy, it needs to be static, and we need to pass the UIWindow to NVC. `UIApplication.shared.keyWindow` returns the CPWindow here so it would have to be a somewhat clumsy workaround.

Reply:
oof. I would strongly prefer that we not reach out to the global like that, preferring to pass it in as an initializer or method argument